src/lib/components/stepper/components/transact-step.svelte

```diff
@@ -318,6 +325,121 @@
+                type: 'uint256',
+              },
+              {
+                name: 'nonce',
+                type: 'uint256',
+              },
+              {
+                name: 'deadline',
+                type: 'uint256',
+              },
+            ],
+          };
+
+          const { data, to } = executingTx.transaction;
+          assert(data, 'Expected data to be defined');
+
+          const { provider } = get(walletStore);
+          const caller = new Contract(networkConfig.contracts.CALLER, callerAbi, provider);
+
+          const nonce = await caller.nonce(address);
+
+          const payload = {
+            sender: address,
+            target: to,
+            data,
+            value: 0,
+            nonce: Number(nonce),
+            // 1 hour in seconds
+            deadline: Math.floor(Date.now() / 1000) + 3600,
+          };
+
+          const sig = await signer.signTypedData(domain, types, payload);
+
+          updateTransactionTimelineStatus(executingTx, {
+            status: 'pending',
+            message: 'Submitting transaction',
+          });
+
+          const gaslessCallRes = await fetch('/api/gasless/call', {
+            method: 'POST',
+            body: JSON.stringify({
+              targetContractAddress: to,
+              callData: data,
+              payload,
+              eip712Signature: sig,
+            }),
+          });
+          const body = await gaslessCallRes.json();
+          const { taskId } = body;
+
+          const gaslessCallExpectation = await expect(
+            async () => {
+              const res = await fetch(`/api/gasless/track/${taskId}`);
+              if (!res.ok) throw new Error('Failed to track gasless owner update task');
+
+              const { task } = await res.json();
+              assert(typeof task === 'object', 'Invalid task');
+              const { taskState, transactionHash } = task;
+              assert(typeof taskState === 'string', 'Invalid task state');
+
+              if (transactionHash) {
+                updateTransactionTimelineStatus(executingTx, {
+                  status: 'pending',
+                  message: 'Waiting for confirmation',
+                  txUrl: networkConfig.explorer.linkTemplate(transactionHash, network.name),
+                });
+              }
+
+              return { taskState, task };
+            },
+            ({ taskState, task }) => {
+              switch (taskState) {
+                case 'ExecSuccess':
+                  return true;
+                case 'Cancelled':
+                  throw new Error(`Gasless transaction failed: ${JSON.stringify(task)}`);
+                default:
+                  return false;
+              }
+            },
+            600000,
+            2000,
+          );
+
+          if (gaslessCallExpectation.failed) {
+            throw new Error('The gasless call did not resolve within the expected timeframe.');
+          }
+
+          break;
+        }
+
```

Comment:
And just noticing the `<script>` here seems long. Is there an opportunity to break it up into bite-sized chunks?

Reply:
#1369